shared/common-adapters/markdown/react.js

```diff
@@ -261,23 +261,23 @@ const reactComponentsForMarkdownType = {
     return <KbfsPath escapedPath={node.content} key={state.key} allowFontScaling={state.allowFontScaling} />
   },
   link: (node, output, state) => {
-    let url = node.content
+    const {protocol, afterProtocol, spaceInFront} = node
+    const rawURL = protocol + afterProtocol
+    const url = (protocol || 'http://') + afterProtocol
+    const title = state.targetURLInLinkTitle ? `${rawURL} (goes to ${url})` : rawURL
 
-    if (!url.match(/^https?:\/\//)) {
-      url = `http://${node.content}`
-    }
     return (
       <React.Fragment key={state.key}>
-        {node.spaceInFront}
+        {spaceInFront}
         <Text
           className="hover-underline"
           type="BodyPrimaryLink"
           style={Styles.collapseStyles([linkStyle, state.styleOverride.link])}
-          title={node.content}
+          title={title}
```

Comment:
this changes the tooltip for storybook vs. non-storybook, but I don't know of any other way to make breakages in the URL parsing show up in storyshots...